jax/_src/compiler.py

```diff
@@ -328,9 +328,21 @@ def compile_or_get_cached(
       module_name, cache_key, compile_options, backend)
   cache_retrieval_time = time.monotonic() - cache_retrieval_start
 
+  hit_log_priority = (logging.WARNING if config.log_compiles.value
+                      else logging.DEBUG)
+  cache_hit_msg = "Persistent compilation cache hit for '%s'"
+  log_cache_hit = lambda: logger.log(hit_log_priority, cache_hit_msg, module_name)
```

Comment:
This function (`compile_or_get_cached`) is already quite long, so I suggest defining `log_cache_hit` and `log_cache_miss` as separate standalone functions that are called inside here. Also maybe rename `log_persistent_cache_hit` etc. for clarity.